lib/plugins/aws/provider/awsProvider.js

```diff
@@ -251,8 +251,85 @@ class AwsProvider {
             },
           },
         },
-        // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8014
-        resources: { type: 'object' },
+        resources: {
+          properties: {
+            AWSTemplateFormatVersion: {
+              type: 'string',
+            },
+            Description: {
+              type: 'string',
+            },
+            Metadata: {
+              type: 'object',
+            },
+            Parameters: {
+              type: 'object',
+            },
+            Mappings: {
+              type: 'object',
+            },
+            Conditions: {
+              type: 'object',
+            },
+            Transform: {
+              type: 'object',
+            },
+            // Not replicating the full JSON schema from https://s3.amazonaws.com/cfn-resource-specifications-us-east-1-prod/schemas/2.15.0/all-spec.json
+            // as that gets into the specifics for each resource type.
+            //
+            // The only required attribute is `Type`; `Properties` and other common attributes are optional.
+            // See also https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html
+            Resources: {
+              type: 'object',
+              patternProperties: {
+                '^[a-zA-Z0-9]{1,255}$': {
+                  type: 'object',
+                  properties: {
+                    Type: { type: 'string' },
+                    Properties: { type: 'object' },
+                    CreationPolicy: { type: 'object' },
+                    DeletionPolicy: { type: 'string' },
+                    DependsOn: { type: 'array', items: { type: 'string' } },
+                    Metadata: { type: 'object' },
+                    UpdatePolicy: { type: 'object' },
+                    UpdateReplacePolicy: { type: 'string' },
+                  },
+                },
+              },
+              additionalProperties: false,
+            },
+            extensions: {
+              type: 'object',
+              patternProperties: {
+                // names have the same restrictions as CloudFormation Resources section
+                '^[a-zA-Z0-9]{1,255}$': {
+                  type: 'object',
+                  // this lists the supported properties, other properties are "Not supported. An error will be thrown
+                  // if you try to extend an unsupported attribute."
+                  properties: {
+                    // this is different than the above schema for `Resources`, which allows the `Type` attribute.
+                    // extensions are explicitly meant to extend the definition of existing resources.
+                    Properties: { type: 'object' },
+                    CreationPolicy: { type: 'object' },
+                    DeletionPolicy: { type: 'string' },
+                    DependsOn: { type: 'array', items: { not: { type: 'null' } } },
```

Comment:
We should allow just string in `DependsOn`

Having this secluded to definition (as proposed [here](https://github.com/serverless/serverless/pull/8139/files#r478314890)) will ensure that definition in both cases are matched, and eventual errors are fixed for both locations.

Reply:
I've fixed the `DependsOn` so that it is an array of strings. Unfortunately I'm not as adept with JSON Schema as I'd like to be so I'm not 100% sure what the definition should look like for this -- your proposal showed what the uses might look like but not the definition itself.